app/lib/search/backend.dart

```diff
@@ -138,10 +148,83 @@ class SearchBackend {
   }
 }
 
+class PopularityStorage {
+  final Storage storage;
+  final Bucket bucket;
+
+  final Map<String, double> _values = {};
+
+  PopularityStorage(this.storage, this.bucket);
+
+  double lookup(String package) => _values[package];
+
+  Future init() async {
+    await fetch();
+    new Timer.periodic(const Duration(days: 1), (_) {
+      fetch();
+    });
+  }
+
+  Future fetch() async {
+    _logger.info('Updating popularity storage...');
+    List<BucketEntry> entries;
+    try {
+      entries = await bucket.list().toList();
+    } catch (e, st) {
+      _logger.severe('Unable to list popularity bucket', e, st);
+      return;
+    }
+    final List<String> names = entries
+        .where((entry) => entry.isObject)
+        .map((entry) => entry.name)
+        .toList();
+    if (names.isEmpty) {
+      _logger.warning('No popularity data found!');
+      return;
+    }
+    // Try to load the available snapshots in reverse order (latest first).
+    names.sort();
+    for (String selected in names.reversed) {
+      _logger.info('Loading popularity data: $selected');
+      try {
+        Stream<List<int>> stream = bucket.read(selected);
+        if (selected.endsWith('.gz')) {
+          stream = stream.transform(_gzip.decoder);
+        }
+        final String json = await stream.transform(UTF8.decoder).join();
+        final Map map = JSON.decode(json);
+        _updateLatest(map);
+        _logger.info('Popularity data loaded successful: $selected');
+        return;
+      } catch (e, st) {
+        _logger.severe('Unable to load popularity data: $selected', e, st);
+      }
+    }
+    return;
+  }
+
+  void _updateLatest(Map raw) {
+    final Map<String, int> rawTotals = {};
+    final List items = raw['items'];
+    for (Map item in items) {
+      final String package = item['pkg'];
+      final int devVotes = int.parse(item['votes_dev']);
+      final int directVotes = int.parse(item['votes_direct']);
+      final int totalVotes = int.parse(item['votes_total']);
+      final int finalVotes = directVotes * 25 + devVotes * 5 + totalVotes;
```

Comment:
I don't know what these votes concretely are, since there hasn't been any code reviews for the generation of this data. But the `+ totalVotes` could be much higher than `directvotes * 25 + devVotes * 5` if the `votes_total` are transitive votes.